clang/include/clang/Frontend/Utils.h

```diff
@@ -87,6 +88,9 @@ class DependencyCollector {
                                   bool IsSystem, bool IsModuleFile,
                                   bool IsMissing);
 
+  /// @return the PPCallback this collector added to the Preprocessor.
+  virtual PPCallbacks *getPPCallbacks() { return nullptr; };
```

Comment:
Do we need this new interface? It's only implemented by single derived class that we have available in the spot where we call `getPPCallbacks()`.

Reply:
Can we keep it downstream for the time being? The default implementation is incorrect, and I'd like to see it in the full context anyway.